librz/bin/format/elf/elf_segments.c

```diff
@@ -56,7 +56,7 @@ static bool get_elf_segment(ELFOBJ *bin, RzBinElfSegment *segment, ut64 offset,
 
 	segment->is_valid = verify_phdr_entry(bin, &segment->data);
 	if (!segment->is_valid) {
-		rz_assert_log(RZ_LOGLVL_WARN, "Invalid segment %zu at 0x%" PFMT64x "\n", pos, offset);
+		RZ_LOG_INFO("Invalid segment %zu at 0x%" PFMT64x "\n", pos, offset);
```

Comment:
@ret2libc note for me to check this.

Reply:
this was already merged here: https://github.com/rizinorg/rizin/pull/1231

Reply:
yes but why was this needed? It seemed alright to have it as a WARNING, no? What was the problem?

Reply:
just tests failing because the ASAN tests have stderr redirected to stdout.

Reply:
https://github.com/rizinorg/rizin/runs/2868460639
Plus that information is more at info level than warning, because otherwise you may get spammed by rizin.

Reply:
@08A apparently the kernel is able to load tiny.elf even if p_offset + p_fsize is bigger than the file size, so I am not sure the second check done in `verify_phdr_entry` is correct.
```C
	Elf_(Off) end_off;
	if (!Elf_(rz_bin_elf_add_off)(&end_off, entry->p_offset, entry->p_filesz) || end_off > bin->size) {
		return false;
	}
```

Reply:
you can point this out in the mm chat.

Reply:
> [rizinorg/rizin/runs/2868460639](https://github.com/rizinorg/rizin/runs/2868460639)
> Plus that information is more at info level than warning, because otherwise you may get spammed by rizin.

What is the default log level? I classified those information as warning because the user should be notified that the loaded sections or segments seems to be invalid.

Reply:
> @08A apparently the kernel is able to load tiny.elf even if p_offset + p_fsize is bigger than the file size, so I am not sure the second check done in `verify_phdr_entry` is correct.
> 
> ```c
> 	Elf_(Off) end_off;
> 	if (!Elf_(rz_bin_elf_add_off)(&end_off, entry->p_offset, entry->p_filesz) || end_off > bin->size) {
> 		return false;
> 	}
> ```

Tiny elf is a corner case. The verification check add more constraint than the kernel, because the idea is to notify the user when the segment or sections seems to be invalid.

Reply:
> just tests failing because the ASAN tests have stderr redirected to stdout.

Shouldn't asan test be redirected to a file? Because some time warning just indicate important information that the user should watch.

Reply:
Indeed I'm not sure why this was converted to INFO... I think converting the log to info just "hides" a problem.

@08A does it break anything if tiny.elf has `segment->is_valid=false`?

Reply:
I test on tiny.elf, and it doesn't seems to break anything (tiny.elf doesn't have any symbols, relocs, ...).

Reply:
because it is indeed an INFO log. we do not need to spam when it fails to load something.

Reply:
@08A ok in this case it does not break anything, but i'm not sure it is alright to set `segment->isvalid` to false if the kernel is actually able to load such a segment. if the kernel can load it, it means the segment is valid IMO.

Reply:
Tiny elf is a corner in maybe 90% of the case if the offset + filesize > bin->size the segments is invalid. The idea isn't to hold the absolute truth but to be good enough. In the future a configuration variable to disable the validity analysis could be created.

Reply:
> in maybe 90% of the case if the offset + filesize > bin->size the segments is invalid

Are they really invalid? Maybe they are only if the `offset` is bigger than bin->size but not if offset + filesize > bin->size... Just guessing from the kernel behaviour I see.

Reply:
I don't know. All those verification are just to notify the user that something seems wrong.